ssl/statem/extensions_cust.c

```diff
@@ -293,31 +318,58 @@ int custom_exts_copy(custom_ext_methods *dst, const custom_ext_methods *src)
             return 0;
         dst->meths_count = src->meths_count;
 
-        for (i = 0; i < src->meths_count; i++) {
-            custom_ext_method *methsrc = src->meths + i;
-            custom_ext_method *methdst = dst->meths + i;
+        for (i = 0; i < src->meths_count; i++)
+            custom_ext_copy_old_cb(&dst->meths[i], &src->meths[i], &err);
+    }
 
-            if (methsrc->add_cb != custom_ext_add_old_cb_wrap)
-                continue;
+    if (err) {
+        custom_exts_free(dst);
+        return 0;
+    }
 
-            /*
-             * We have found an old style API wrapper. We need to copy the
-             * arguments too.
-             */
+    return 1;
+}
 
-            if (err) {
-                methdst->add_arg = NULL;
-                methdst->parse_arg = NULL;
-                continue;
-            }
+/* Copy custom extensions that were set on connection */
+int custom_exts_copy_conn(custom_ext_methods *dst,
+                          const custom_ext_methods *src)
+{
+    size_t i;
+    int err = 0;
 
-            methdst->add_arg = OPENSSL_memdup(methsrc->add_arg,
-                                              sizeof(custom_ext_add_cb_wrap));
-            methdst->parse_arg = OPENSSL_memdup(methsrc->parse_arg,
-                                            sizeof(custom_ext_parse_cb_wrap));
+    if (src->meths_count > 0) {
+        size_t meths_count = 0;
+
+        for (i = 0; i < src->meths_count; i++)
+            if (src->meths[i].ext_flags & SSL_EXT_FLAG_CONN)
```

Comment:
Our style is generally this:
```suggestion
            if ((src->meths[i].ext_flags & SSL_EXT_FLAG_CONN) != 0)
```